src/hotspot/share/opto/graphKit.cpp

```diff
@@ -1019,15 +1019,20 @@ bool GraphKit::compute_stack_effects(int& inputs, int& depth) {
     code = method()->java_code_at_bci(bci() + 1);
   }
 
-  BasicType rtype = T_ILLEGAL;
-  int       rsize = 0;
-
   if (code != Bytecodes::_illegal) {
     depth = Bytecodes::depth(code); // checkcast=0, athrow=-1
+  }
+
+  auto rsize = [&]() {
+    BasicType rtype = T_ILLEGAL;
```

Comment:
Rather than initializing `rtype` to a dummy value and then almost immediately assigning `rtype`, just initialize `rtype` properly to begin with.